### ISSUE TITLE
Make conversion of all data fields to float OPTIONAL

### ISSUE THREAD

**Issue by weidinger-c** [CloudCompare/CloudCompare#2045]
The las standard allows for data fields like double, uint64 etc. 
Unfortunately after importing the pointclouds in CloudCompare, all data is converted to float values, which results in overflow of the values. Hence, displaying these data types does not work. 
Is there any option to keep the original data types (at least support the ones in the offical LAS standard)?

**Comment by dgirardeau** [comment]
Is it with the latest version? @tmontaigu any idea?

**Comment by weidinger-c** [comment]
@dgirardeau 
One example: I have point_ids of uint64 that are unique and monotic increasing. The coloring makes no sense and also the values are of type float (and there overflow)
![image](https://github.com/user-attachments/assets/05ea8cd4-b296-4915-8784-129b6b027f3b)

**Comment by tmontaigu** [comment]
`Is there any option to keep the original data types (at least support the ones in the offical LAS standard)?`
this is already the case, your point_id thing is not standard

And since the CC stores as float I don't know what could be done since here a shift mechanism is not going to help I guess

**Comment by weidinger-c** [comment]
what do you mean with not standard? The LAS standard allows for attibutes of the following types:
![image](https://github.com/user-attachments/assets/1920ece2-40a4-4f3c-a332-48a352a11518)

**Comment by dgirardeau** [comment]
We could do as with GPS time (with a shift).

But I understand now better the issue, it's trying to store a non-standard field in a CloudCompare's Scalar Field.

**Comment by weidinger-c** [comment]
Yes LAS allows for attributes added as VLR, which is oftenly used. It works for most data types, but as shown not for datatypes like uint64 etc

**Comment by weidinger-c** [comment]
Yes, that is exactly what I meant, so I guess the underlying structure of storing everthing as float is at fault.

**Comment by dgirardeau** [comment]
Well, the aim was too minimize the size in memory. And floats work in most of the cases so far ... It has just not been designed for what you are trying to do. Making the storage type of scalar fields dynamic would have quite a big impact on the existing code-base. It would be more realistic to add new types of Scalar fields, but they wouldn't be managed by most of the existing tools. At least they would be preserved.

**Comment by weidinger-c** [comment]
Ok thanks for the info.

**Comment by tmontaigu** [comment]
Yeah I read to fast the 'storage' and though it meant storage on the disk not in memory

Being able to store (in memory) and visualize types other than float would already be a big step , given how many times we have conplaints  with GpsTime for example 😬
